GeneratorInterface/Herwig7Interface/src/Herwig7Interface.cc

```diff
@@ -0,0 +1,339 @@
+    return;
+  }
+  catch ( const char* what ) {
+    edm::LogError("Herwig7Interface") << ": caught exception: "
+	      << what << "\n";
+    return;
+  }
+  catch ( ... ) {
+    edm::LogError("Herwig7Interface") << ": Unknown exception caught.\n";
+    return;
+  }
+
+
+
+}
+
+
+bool Herwig7Interface::initGenerator()
+{
+	if ( HwUI_->runMode() == Herwig::RunMode::RUN) {
+		edm::LogInfo("Herwig7Interface") << "Starting EventGenerator initialization";
+		callHerwigGenerator();
+		edm::LogInfo("Herwig7Interface") << "EventGenerator initialized";
+
+		// Skip events
+		for (unsigned int i = 0; i < skipEvents_; i++) {
+			flushRandomNumberGenerator();
+			eg_->shoot();
+			edm::LogInfo("Herwig7Interface") << "Event discarded";
+		}
+
+		return true;
+
+	} else {
+		edm::LogInfo("Herwig7Interface") << "Stopped EventGenerator due to missing run mode.";
+		return false;
+/*
+		throw cms::Exception("Herwig7Interface")
+			<< "EventGenerator could not be initialized due to wrong run mode!" << endl;
+*/
+	}
+
+}
+
+void Herwig7Interface::flushRandomNumberGenerator()
+{
+	/*ThePEG::RandomEngineGlue *rnd = randomEngineGlueProxy_->getInstance();
+
+	if (!rnd)
+		edm::LogWarning("ProxyMissing")
+			<< "ThePEG not initialised with RandomEngineGlue.";
+	else
+		rnd->flush();
+      */
+}
+
+auto_ptr<HepMC::GenEvent> Herwig7Interface::convert(
+					const ThePEG::EventPtr &event)
+{
+	return std::auto_ptr<HepMC::GenEvent>(
+		ThePEG::HepMCConverter<HepMC::GenEvent>::convert(*event));
+}
+
+
+
+
+double Herwig7Interface::pthat(const ThePEG::EventPtr &event)
+{
+	using namespace ThePEG;
+
+	if (!event->primaryCollision())
+		return -1.0;
+
+	tSubProPtr sub = event->primaryCollision()->primarySubProcess();
+	TmpTransform<tSubProPtr> tmp(sub, Utilities::getBoostToCM(
+							sub->incoming()));
+
+	double pthat = (*sub->outgoing().begin())->momentum().perp() / ThePEG::GeV;
+	for(PVector::const_iterator it = sub->outgoing().begin();
+	    it != sub->outgoing().end(); ++it)
+		pthat = std::min<double>(pthat, (*it)->momentum().perp() / ThePEG::GeV);
+
+	return pthat;
+}
+
+
+
+
+void Herwig7Interface::createInputFile(const edm::ParameterSet &pset)
+{
+	/* Initialize the input config for Herwig from
+	 * 1. the Herwig7 config files
+	 * 2. the CMSSW config blocks
+	 * Writes them to an output file which is read by Herwig
+	*/
+
+	stringstream logstream;
+
+
+	// Contains input config passed to Herwig
+	stringstream herwiginputconfig;
+
+	// Define output file to which input config is written, too, if dumpConfig parameter is set. 
+	// Otherwise use default file HerwigConfig.in which is read in by Herwig
+	ofstream cfgDump;
+	cfgDump.open(dumpConfig_.c_str(), ios_base::app);
+	
+
+
+	// Read Herwig config files as input
+	vector<string> configFiles = pset.getParameter<vector<string> >("configFiles");
+	// Loop over the config files
+	for(vector<string>::iterator iter = configFiles.begin(); iter != configFiles.end(); ++iter) {
```

Comment:
please use a range-based loop here ( for ( const auto & iter : configFiles ) ..)